src/test/java/seedu/address/logic/commands/CreateTagCommandTest.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code CreateTagCommand}.
+ */
+public class CreateTagCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validTagName_success() {
+        CreateTagCommand createTagCommand = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        String expectedMessage = String.format(CreateTagCommand.MESSAGE_CREATE_TAG_SUCCESS, VALID_TAGNAME_NEIGHBOURS);
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addTag(new Tag(VALID_TAGNAME_NEIGHBOURS));
+        assertCommandSuccess(createTagCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_sameTagNameDifferentCase_throwsCommandException() {
+        Tag newTag = new Tag(VALID_TAGNAME_NEIGHBOURS);
+        model.addTag(newTag);
+        CreateTagCommand createTagCommand1 = new CreateTagCommand("frIenDS");
+        String expectedMessage = CreateTagCommand.MESSAGE_DUPLICATE_TAG;
+        assertCommandFailure(createTagCommand1, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_tagDoesNotExistInModel_success() {
+        Tag newTag = new Tag(VALID_TAGNAME_NEIGHBOURS);
+        CreateTagCommand createTagCommand = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        String expectedMessage = String.format(CreateTagCommand.MESSAGE_CREATE_TAG_SUCCESS, VALID_TAGNAME_NEIGHBOURS);
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addTag(newTag);
+        assertCommandSuccess(createTagCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_tagExistsInModel_throwsCommandException() {
+        Tag newTag = new Tag(VALID_TAGNAME_NEIGHBOURS);
+        model.addTag(newTag);
+        CreateTagCommand createTagCommand = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        String expectedMessage = CreateTagCommand.MESSAGE_DUPLICATE_TAG;
+        assertCommandFailure(createTagCommand, model, expectedMessage);
+    }
+
+    @Test
+    public void equals() {
+        CreateTagCommand createTagFirstCommand = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        CreateTagCommand createTagSecondCommand = new CreateTagCommand(VALID_TAGNAME_COLLEAGUES);
+        // same object -> returns true
+        assertTrue(createTagFirstCommand.equals(createTagFirstCommand));
+
+        // same values -> returns true
+        CreateTagCommand createTagFirstCommandCopy = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        assertTrue(createTagFirstCommand.equals(createTagFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(createTagFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(createTagFirstCommand.equals(null));
+
+        // different person -> returns false
```

Comment:
I think you meant to write `different tag`

Reply:
Yes haha thanks for catching it! Sharp eyes ✔️